src/test.ts

```diff
@@ -84,6 +86,11 @@ export function runTest(testDirectory: string, rulesDirectory?: string | string[
         };
         compilerOptions = ts.parseJsonConfigFileContent(config, parseConfigHost, testDirectory).options;
     }
+    // TODO remove in v6.0.0
+    if (tslintConfig !== undefined && tslintConfig.linterOptions !== undefined && tslintConfig.linterOptions.typeCheck === true) {
```

Comment:
`linterOptions.typecheck` was never public; we can remove it immediately. https://github.com/palantir/tslint/issues/1445#issuecomment-236330061